### ISSUE TITLE
TensorFlow 1.14 custom estimator model slower than TensorFlow 1.13

### ISSUE THREAD

**Issue by dwangnuro** [tensorflow/tensorflow#30967]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
No
- TensorFlow installed from (source or binary):
Binary
- TensorFlow version (use command below):
TensorFlow 1.14
- Python version:
Python 2.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
CUDA 10, CuDNN 7.4.2
- GPU model and memory:
GeForce GTX 1080Ti

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

I was trying to upgrade my tensorflow from 1.13 to tensorflow 1.14. I noticed that the training speed for my models dropped a lot.

Then I tried the tensorflow custom estimator and reproduced the problem.
https://github.com/tensorflow/models/blob/master/samples/core/get_started/custom_estimator.py

This is the speed with tf 1.13.
INFO:tensorflow:loss = 2.3727872, step = 0
INFO:tensorflow:global_step/sec: 293.209
INFO:tensorflow:loss = 0.17633303, step = 100 (0.341 sec)
INFO:tensorflow:global_step/sec: 337.629
INFO:tensorflow:loss = 0.09756016, step = 200 (0.296 sec)
INFO:tensorflow:global_step/sec: 342.554
INFO:tensorflow:loss = 0.10544465, step = 300 (0.292 sec)
INFO:tensorflow:global_step/sec: 342.051
INFO:tensorflow:loss = 0.049246334, step = 400 (0.292 sec)
INFO:tensorflow:global_step/sec: 347.739
INFO:tensorflow:loss = 0.09701028, step = 500 (0.288 sec)
INFO:tensorflow:global_step/sec: 347.722
INFO:tensorflow:loss = 0.063723646, step = 600 (0.287 sec)
INFO:tensorflow:global_step/sec: 329.121
INFO:tensorflow:loss = 0.045053717, step = 700 (0.304 sec)
INFO:tensorflow:global_step/sec: 295.816
INFO:tensorflow:loss = 0.061943192, step = 800 (0.338 sec)
INFO:tensorflow:global_step/sec: 295.844
INFO:tensorflow:loss = 0.029347159, step = 900 (0.338 sec)

This is the speed with tf 1.14.
I0722 22:54:59.609863 139637823923968 basic_session_run_hooks.py:262] loss = 1.0876435, step = 0
I0722 22:54:59.962240 139637823923968 basic_session_run_hooks.py:692] global_step/sec: 283.369
I0722 22:54:59.963597 139637823923968 basic_session_run_hooks.py:260] loss = 0.13201241, step = 100 (0.354 sec)
I0722 22:55:00.268446 139637823923968 basic_session_run_hooks.py:692] global_step/sec: 326.584
I0722 22:55:00.270031 139637823923968 basic_session_run_hooks.py:260] loss = 0.09678831, step = 200 (0.306 sec)
I0722 22:55:00.601743 139637823923968 basic_session_run_hooks.py:692] global_step/sec: 299.99
I0722 22:55:00.602967 139637823923968 basic_session_run_hooks.py:260] loss = 0.07743741, step = 300 (0.333 sec)
I0722 22:55:00.933660 139637823923968 basic_session_run_hooks.py:692] global_step/sec: 301.274
I0722 22:55:00.935039 139637823923968 basic_session_run_hooks.py:260] loss = 0.08071006, step = 400 (0.332 sec)
I0722 22:55:01.271045 139637823923968 basic_session_run_hooks.py:692] global_step/sec: 296.425
I0722 22:55:01.272447 139637823923968 basic_session_run_hooks.py:260] loss = 0.10450166, step = 500 (0.337 sec)
I0722 22:55:01.605127 139637823923968 basic_session_run_hooks.py:692] global_step/sec: 299.334
I0722 22:55:01.606523 139637823923968 basic_session_run_hooks.py:260] loss = 0.06578785, step = 600 (0.334 sec)
I0722 22:55:01.946224 139637823923968 basic_session_run_hooks.py:692] global_step/sec: 293.167
I0722 22:55:01.947628 139637823923968 basic_session_run_hooks.py:260] loss = 0.09801171, step = 700 (0.341 sec)
I0722 22:55:02.284446 139637823923968 basic_session_run_hooks.py:692] global_step/sec: 295.649
I0722 22:55:02.285842 139637823923968 basic_session_run_hooks.py:260] loss = 0.05814831, step = 800 (0.338 sec)
I0722 22:55:02.621150 139637823923968 basic_session_run_hooks.py:692] global_step/sec: 297.002
I0722 22:55:02.622456 139637823923968 basic_session_run_hooks.py:260] loss = 0.056445364, step = 900 (0.337 sec)

**Describe the expected behavior**
I would expect tensorflow has similar or better performance after the upgrade.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
https://github.com/tensorflow/models/blob/master/samples/core/get_started/custom_estimator.py

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by gadagashwini** [comment]
@dwangnuro This question is better asked on [tenosrflow/model](https://github.com/tensorflow/models/issues) repository for faster resolution. Thanks!

**Comment by dwangnuro** [comment]
@gadagashwini Why is that? It is the same demo code but became slower after upgrade. And I saw my other code became 50% more slower after upgrade. Is it a regression on tensorflow?

**Comment by gadagashwini** [comment]
@dwangnuro As this is more related to TF models, please post this in [TF model](https://github.com/tensorflow/models/issues) repo. Thanks!

**Comment by dwangnuro** [comment]
@gadagashwini I think there are some misunderstanding. In general, I noticed some of my model become slower after upgrade to tf 1.14. And the model in this issue is just an example that shows tf 1.14 is slower than tf 1.13. It is a performance issue. It isn't the model in the TF model become slower.

**Comment by gadagashwini** [comment]
@dwangnuro If some of your custom model became slower after upgrading to 1.14, might be regression issue in 1.14 version. Will take a look. Thanks!

**Comment by ymodak** [comment]
@dwangnuro I see that time is almost similar. This doesn't look like a regression issue. Perhaps you can use profiling tool and investigate which TF operation consumes longer time than usual to execute. This will help us to troubleshoot quicker.  We will reopen the issue when new information is available. Thanks!

**Comment by dwangnuro** [comment]
On a different RNN model that I'm working on. The training speed is 50% slower. I tried the profiling tool but TF 1.14 and TF 1.13 generates very different profiling result.
1. TF 1.14 put data ops into the graph so there are a lot of profiling result for them which are not in the TF 1.13 profiling graph. I cannot tell if it is cause for the slowness.
2. In TF 1.13, there's only one CPU compute section. In TF 1.14, I saw two CPU compute section. It seems that every op runs on the GPU also appears in one of the CPU compute section. Is it intended?
